js/packages/openinference-vercel/src/OpenInferenceSpanProcessor.ts

```diff
@@ -1,26 +1,112 @@
-import { ReadableSpan, SpanProcessor } from "@opentelemetry/sdk-trace-base";
-import { safelyGetOpenInferenceAttributes } from "./utils";
-import { ReadWriteSpan } from "./types";
+import {
+  BatchSpanProcessor,
+  BufferConfig,
+  ReadableSpan,
+  SimpleSpanProcessor,
+  SpanExporter,
+} from "@opentelemetry/sdk-trace-base";
+import { addOpenInferenceAttributesToSpan, shouldExportSpan } from "./utils";
 
-export class OpenInferenceSpanProcessor implements SpanProcessor {
-  async forceFlush(): Promise<void> {
-    // no-op
+/**
+ * Extends {@link SimpleSpanProcessor} to support OpenInference attributes.
+ * This processor enhances spans with OpenInference attributes before exporting them.
+ * It can be configured to selectively export only OpenInference spans or all spans.
+ * @extends {SimpleSpanProcessor}
+ *
+ * @example
+ * ```typescript
+ * const exporter = new OTLPTraceExporter();
+ * const processor = new OpenInferenceSimpleSpanProcessor({
+ *   exporter,
+ *   onlyExportOpenInferenceSpans: true,
+ * });
+ * tracerProvider.addSpanProcessor(processor);
+ * ```
+ */
+export class OpenInferenceSimpleSpanProcessor extends SimpleSpanProcessor {
+  private readonly onlyExportOpenInferenceSpans: boolean;
+  constructor({
```

Comment:
optionally the signature could be constructor(exporter: Exporter, config: ExportConfig)

I guess this design leaves it a bit more open to more parameters...

Reply:
Optionally we could expose this as just something called span filter?

Reply:
The more I think about it I think this is the safest option